src/pages/SingleProductPage.js

```diff
@@ -1,22 +1,95 @@
-import React, { useEffect } from 'react'
-import { useParams, useHistory } from 'react-router-dom'
-import { useProductsContext } from '../context/products_context'
-import { single_product_url as url } from '../utils/constants'
-import { formatPrice } from '../utils/helpers'
+import React, { useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useProductsContext } from '../context/products_context';
+import { single_product_url as url } from '../utils/constants';
+import { formatPrice } from '../utils/helpers';
 import {
   Loading,
   Error,
   ProductImages,
   AddToCart,
   Stars,
   PageHero,
-} from '../components'
-import styled from 'styled-components'
-import { Link } from 'react-router-dom'
-
+} from '../components';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
 const SingleProductPage = () => {
-  return <h4>single product page</h4>
-}
+  const { id } = useParams();
+  const idNumber = Number(id); //converted to number
+  console.log(id);
+  //const navigate = useNavigate();
+  const {
+    single_product_loading: loading,
+    single_product_error: error,
+    single_product: product,
+    fetchSingleProduct,
+  } = useProductsContext();
+//but i still cant fetch single product
+  useEffect(() => {
+    fetchSingleProduct(url, idNumber);
+    // eslint-disable-next-line 
+  }, [idNumber]);
+  
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => {
+       // navigate('/');
```

Comment:
Curious to understand why this is commented out? Do we not want to navigate back to home if there was an error? In that case we could probably just remove this whole useEffect which is fine as well. In fact, that would make sense since we already have on line 47 an if statement to handle if there's an error

Reply:
@akosasante I will remove it completely, i commented it out while trying to solve the bug - i was adding and deleting the import of useParams and useNavigate hooks - instead of deleting only useParams i deleted both so the terminal was complaining and i commented it out. this useEffect hook gives a nice feature- redirects user to a home page 3s after there was an error- but i cant make it work- it gives me an error saying export 'useNavigate' (imported as 'useNavigate') was not found in 'react-router-dom'  - i dont know why does it happend- i have a package installed, imported but it doesnt let me use it 
![image](https://github.com/NatachaKey/reactEcommerce/assets/103755090/3c402e25-327d-4b09-a180-aedefa4babbc)
![image](https://github.com/NatachaKey/reactEcommerce/assets/103755090/9ec33649-938a-4692-b3fe-5185ff6bd8d3)

Reply:
Looking at your package-lock.json, you have this version of `react-router-dom`: 5.3.4 (you can also see this if you run the command `npm list react-router-dom`.

Based on the documentation `useNavigate` was introduced in v6 of the React Router package. In v5 which you're using, you instead would use `useHistory`: https://reactrouter.com/en/main/upgrading/v5#use-usenavigate-instead-of-usehistory

```jsx
import { Link, useHistory } from 'react-router-dom';

const history = useHistory();

useEffect(() => {
    if (error) {
      setTimeout(() => {
        history.push('/');
      }, 3000)
    }
});
```